sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -538,6 +539,19 @@ object CollapseRepartition extends Rule[LogicalPlan] {
 }
 
 /**
+ * Collapse Adjacent Window Expression.
+ * - Collapse only if they have the same partition specs and order specs.
+ */
+object CollapseWindow extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
+    case w @ Window(we1, ps1, os1, Window(we2, ps2, os2, grandChild))
+        if ps1.equals(ps2) && os1.equals(os2) =>
+      val newWindowExpressions = ExpressionSet(we1 ++ we2).toSeq.asInstanceOf[Seq[NamedExpression]]
```

Comment:
Ur, it seems my mistake. At first, I thought we had better remove duplicated window expression. I'll fix this right now.